src/DssDirectDepositHub.sol

```diff
@@ -215,6 +212,7 @@ contract DssDirectDepositHub {
         // That's why it converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
         // This module will have an unintended behaviour if rate is changed to some other value.
 
+        EndLike end_ = end;
```

Comment:
Well the `end` variable should be assigned only in the `else` case like before otherwise it is reading the storage always when it is not needed.